ui/pages/Project/components/FamilyTable/FamilyTable.jsx

```diff
@@ -50,38 +55,47 @@ class FamilyTableRow extends React.PureComponent {
     ))
   }
 
+  handleOnScreen = () => {
+    this.setState({ isVisible: true })
+  }
+
   render() {
     const {
       familyGuid, showVariantDetails, detailFields, noDetailFields, tableName, showDetails: initialShowDetails,
     } = this.props
-    const { showDetails } = this.state
+    const { showDetails, isVisible } = this.state
     const showFamilyDetails = showDetails === null ? initialShowDetails : showDetails
     return (
       <Table.Row>
         <OverflowCell>
-          <FamilyDetail
-            key={familyGuid}
-            familyGuid={familyGuid}
-            showFamilyPageLink
-            showVariantDetails={showVariantDetails}
-            tableName={tableName}
-            fields={showFamilyDetails ? detailFields : noDetailFields}
-            compact={!showFamilyDetails}
-            disableEdit={!showFamilyDetails}
-            annotation={detailFields && noDetailFields && <ToggleIcon rotated={showFamilyDetails ? undefined : 'counterclockwise'} onClick={this.toggle} />}
-            showIndividuals={showFamilyDetails}
-          />
+          <Visibility fireOnMount onOnScreen={this.handleOnScreen}>
```

Comment:
this causes react to only render a row in the family table if it is scrolled into view (including if the page is scrolled so it would already be visible). For projects with > 500 families, this can drop the javascript rendering time for the page by over 5 seconds, and the rendering when scrolled into view happens basically instantaneously and does not look bad to the end user